packages/shared/lib/core/profile-manager/api/events/transaction-progress.event.ts

```diff
@@ -0,0 +1,83 @@
+import { selectedAccountId } from '@core/account'
+import { get } from 'svelte/store'
+import { isLedgerProfile } from '@core/profile'
+import { closePopup, openPopup } from '@lib/popup'
+import { ledgerDeviceStatus, ledgerMintNativeTokenConfirmationProps, ledgerSendConfirmationProps } from '@core/ledger'
+
+import { TransactionProgressPayload } from '../types'
+
+export function handleTransactionProgressEvent(accountId: string, payload: TransactionProgressPayload): void {
+    if (get(isLedgerProfile)) {
+        if (get(selectedAccountId) === accountId) {
+            handleTransactionProgressInternal(payload)
+        }
+    } else {
+        console.warn('Transaction progress handler unimplemented: ', payload)
+    }
+}
+
+function handleTransactionProgressInternal(
+    payload: TransactionProgressPayload,
+    isDuringOnboarding: boolean = false
```

Comment:
Doesn't make sense to type if you initialize it with a `false`. TS knows infers that it's a boolean

Reply:
Fair enough, but for more complicated parameters like a profile manager, it is also inferred by TS but without the type there it's not dev friendly. Although not the same as a `boolean`, the inconsistency bothers me :sweat_smile: